src/ADAL.PCL/AdalHttpClient.cs

```diff
@@ -132,10 +155,11 @@ public AdalHttpClient(string uri, CallState callState)
             return typedResponse;
         }
 
+
         private bool isDeviceAuthChallenge(string endpointType, IHttpWebResponse response, bool respondToDeviceAuthChallenge)
         {
             return PlatformPlugin.DeviceAuthHelper.CanHandleDeviceAuthChallenge &&
-                   respondToDeviceAuthChallenge &&
+                   respondToDeviceAuthChallenge && response!=null &&
```

Comment:
also normally i expect new conditions to get added to the end, any reason to do it in the middle?

---
In reply to: [66913931](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/454#discussion_r66913931) [](ancestors = 66913931)